### PR TITLE
docs: update specs for v1.0.0

### PR DESCRIPTION
**Pull request by 0xHansLee**
# Description

This PR includes all the updates of specs for `v1.0.0`.
- multiple challenge system
- security council
- contract upgrade
- etc (reformatting, rewording, and reorganizing)

There is still a lot missing from the current spec, so let's improve it more for future.

### CODE REVIEW

File: specs/challenge.md [hunk 1]
```diff
@@ -3,21 +3,31 @@
 <!-- All glossary references in this file. -->
 
 [g-l1]: glossary.md#layer-1-l1
+
```
Comment (Pangssu) [inline specs/challenge.md:6]: please remove newlines

File: specs/security-council.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Security Council
+
+[g-l1]: glossary.md#layer-1-l1
+
```
Comment (Pangssu) [inline specs/security-council.md:4]: please remove newline

File: specs/challenge.md [hunk 3]
```diff
@@ -110,212 +148,325 @@ You can notice that in each turn, the first element of the segments must be same
...
-  import { keccak256 } from 'ethers/lib/utils';
+import { DataOptions, hexlify } from '@ethersproject/bytes';
+import { Wallet, constants } from 'ethers';
+import { keccak256 } from 'ethers/lib/utils';
 
-  function strip0x(str: string): string {
+function strip0x(str: string): string {
     if (str.startsWith('0x')) {
-      return str.slice(2);
+        return str.slice(2);
```
Comment (Pangssu) [inline specs/challenge.md:166]: We're using 2 spaces as tab width, so it's better to delete the indent you added.

**Review by seolaoh — APPROVED** [review]

**Review by seolaoh — APPROVED** [review]

**Review by kangsorang — APPROVED** [review]
Lgtm

**Review by Pangssu — APPROVED** [review]

**Pull request merged**